src/minmax.rs

```diff
@@ -0,0 +1,273 @@
+
+        for i in self.estimate.iter_mut() {
+            *i = val;
+        }
+        self.estimate[0].value
+    }
+
+    // Check if new measurement updates previous min estimates.
+    pub fn running_min(
+        &mut self, win: Duration, time: Instant, meas: Duration,
+    ) -> Duration {
+        let val = MinmaxSample { time, value: meas };
+
+        let temp = time.duration_since(self.estimate[2].time);
+
+        // if nothing in window or found new min value, reset!
+        if temp.gt(&win) || val.value.le(&self.estimate[0].value) {
+            return self.reset(time, meas);
+        }
+
+        if val.value.le(&self.estimate[1].value) {
+            self.estimate[2] = val;
+            self.estimate[1] = val;
+        } else if val.value.le(&self.estimate[2].value) {
+            self.estimate[2] = val;
+        }
+
+        self.subwin_update(win, time, meas)
+    }
+
+    // Check if new measurement updates previous max estimates.
+    pub fn _running_max(
+        &mut self, win: Duration, time: Instant, meas: Duration,
+    ) -> Duration {
+        let val = MinmaxSample { time, value: meas };
+
+        let temp = time.duration_since(self.estimate[2].time);
+
+        // if nothing in window or found new max value, reset!
+        if val.value.ge(&self.estimate[0].value) || temp.gt(&win) {
+            return self.reset(time, meas);
+        }
+
+        if val.value.ge(&self.estimate[1].value) {
+            self.estimate[2] = val;
+            self.estimate[1] = val;
+        } else if val.value.ge(&self.estimate[2].value) {
+            self.estimate[2] = val
+        }
+
+        self.subwin_update(win, time, meas)
+    }
+
+    fn subwin_update(
+        &mut self, win: Duration, time: Instant, meas: Duration,
+    ) -> Duration {
+        let val = MinmaxSample { time, value: meas };
+
+        let dt = time.duration_since(self.estimate[0].time);
+
+        if dt.gt(&win) {
+            self.estimate[0] = self.estimate[1];
+            self.estimate[1] = self.estimate[2];
+            self.estimate[2] = val;
+
+            let temp = time.duration_since(self.estimate[0].time);
+            if temp.gt(&win) {
+                self.estimate[0] = self.estimate[1];
+                self.estimate[1] = self.estimate[2];
+                self.estimate[2] = val;
+            }
+        } else if self.estimate[1].time.eq(&self.estimate[0].time) &&
+            dt.gt(&win.div_f32(4.0))
+        {
+            self.estimate[2] = val;
+            self.estimate[1] = val;
+        } else if self.estimate[2].time.eq(&self.estimate[1].time) &&
+            dt.gt(&win.div_f32(2.0))
+        {
+            self.estimate[2] = val;
+        }
+        self.estimate[0].value
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn reset_filter() {
+        let mut f = Minmax::new();
+        let now = Instant::now();
+        let rtt = Duration::new(0, 50000000);
```

Comment:
Better use `Duration::from_millis()`, as it's easier to understand what the unit is without having to count the zeros :)

Same below.